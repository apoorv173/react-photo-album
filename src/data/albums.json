[
  {
    "username": 1,
    "albumId": 1,
    "albumTitle": "quidem molestiae enim",
    "color": "fff"
  },
  {
    "username": 1,
    "albumId": 2,
    "albumTitle": "sunt qui excepturi placeat culpa",
    "color": "111"
  }
]