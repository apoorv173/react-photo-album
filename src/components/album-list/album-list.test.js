import React from "react";
import AlbumList from "./album-list";
import { shallow, mount } from "enzyme";
import albums from '../../data/albums.json';
import { Link } from "react-router-dom";
import {MemoryRouter} from "react-router-dom";

describe("Album List Component", () => {

    const onNextClick = jest.fn();
    const onPrevClick = jest.fn();

    it("Should render albums with user data", () => {
        const tree = shallow(<MemoryRouter><AlbumList albums={albums} /></MemoryRouter>)
        expect(tree).toMatchSnapshot();
    });

    it('should render all the components', () => {
        const wrapper = mount(<MemoryRouter><AlbumList albums={albums} /></MemoryRouter>);
        expect(wrapper.find(Link).length).toBe(2);
        expect(wrapper.find('.row').length).toBe(1);
        expect(wrapper.find('.thumbnail').length).toBe(2);
        expect(wrapper.find('.thumbnail-img').length).toBe(2);
        expect(wrapper.find('.thumbnail-heading').length).toBe(2);
        expect(wrapper.find('.thumbnail-text').length).toBe(2);
        expect(wrapper.find('.layer').length).toBe(2);
    })
});