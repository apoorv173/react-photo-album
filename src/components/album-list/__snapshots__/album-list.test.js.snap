// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album List Component Should render albums with user data 1`] = `ShallowWrapper {}`;
